tests/core/math/test_math_funcs.h

```diff
@@ -0,0 +1,556 @@
+/*************************************************************************/
+/*  test_math_funcs.h                                                    */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_MATH_FUNCS_H
+#define TEST_MATH_FUNCS_H
+
+#include "tests/test_macros.h"
+
+namespace TestMath {
+
+TEST_CASE("[Math] C++ macros") {
+	CHECK(MIN(-2, 2) == -2);
+	CHECK(MIN(600, 2) == 2);
+
+	CHECK(MAX(-2, 2) == 2);
+	CHECK(MAX(600, 2) == 600);
+
+	CHECK(CLAMP(600, -2, 2) == 2);
+	CHECK(CLAMP(620, 600, 650) == 620);
+	// `max` is lower than `min`.
+	CHECK(CLAMP(620, 600, 50) == 50);
+
+	CHECK(ABS(-5) == 5);
+	CHECK(ABS(0) == 0);
+	CHECK(ABS(5) == 5);
+
+	CHECK(SIGN(-5) == doctest::Approx(-1.0));
+	CHECK(SIGN(0) == doctest::Approx(0.0));
+	CHECK(SIGN(5) == doctest::Approx(1.0));
```

Comment:
These should be tested to have exact results, since they are trivial operations with no rounding errors.

Reply:
Done :)